### PR TITLE
Make paper-drawer-panel#rightDrawer default to true in RTL

### LINKED ISSUE #101: Drawer Panel appears on wrong side when using RTL document direction
When using the paper-drawer-panel in a top level body tag with dir="rtl", the drawer panel appears on the same side as when dir="ltr".  We'd expect the panel to switch sides.

### PR DESCRIPTION
**Pull request by danbeam**
possibly fixes https://github.com/PolymerElements/paper-drawer-panel/issues/101

i also thought about creating a

``` js
directionChanged: function() {
  this.rightDrawer = this._isRTL();
},
```

for folks to call if directionality changes, but:
- I think it might be odd for some as a side effect
- `rightDrawer` is already `@public` so this doesn't add much value (authors can do this themselves if they wish)

/cc @blasten @frankiefu

### CODE REVIEW

File: paper-drawer-panel.html [hunk 1]
```diff
@@ -550,6 +552,10 @@
           this._transition = true;
         },
 
+        _isRTL: function() {
+          return window.getComputedStyle(this).direction == 'rtl';
```
Comment (blasten) [inline paper-drawer-panel.html]: `===`

**Comment by blasten** [comment]
Thanks for fixing this @danbeam. The test is still failing, do you know why?

**Comment by danbeam** [comment]
according to @esprehn I'm hitting a bug where calling `getComputedStyle()` outside of the `document` or in a `<template>` returns an empty string rather than (in `direction`'s case) `'ltr'` or `'rtl'`.

**Comment by notwaldorf** [comment]
@danbeam does it help if you set `rightDrawer` in `attached()` instead rather than in the properties block (i.e. to make sure that you're in the document when it matters)

**Comment by blasten** [comment]
would this help?

``` js
var drawer;

setup(function() {
  drawer = fixture('rtl-drawer');
});

test('rtl-drawer defaults rightDrawer to true', function(done) {
   flush(function() {
     expect(drawer.rightDrawer).to.be.true;
     done();
    });
});
```

**Comment by keanulee** [comment]
One of the reasons why tests are failing is because you are overriding `rightDrawer` based on RTL even if it has been explicitly set. One alternative is to have `rightDrawer` default as `null` in the `properties` object, then initialize it in `attached` based on RTL only if it is `null` (and not `false`). Not sure if this is good conventionally, but it works. WDYT @danbeam @notwaldorf @blasten ?

``` javascript
rightDrawer: {
  type: Boolean,
  value: null
}

...

attached: function() {
  if (this.rightDrawer === null) {
    this.rightDrawer = this._isRTL();
  }
}
```

**Comment by danbeam** [comment]
@keanulee that helps and fixes the issue of this:

``` js
var paperDrawerPanel = document.createElement('paper-drawer-panel');
paperDrawerPanel.rightDrawer = ...;
document.body.appendChild(paperDrawerPanel);  // overriden
```

but doesn't pass all tests yet

**Comment by keanulee** [comment]
@danbeam There's only one failing test, and here's the fix :) In test/small-devices.html, change the contents of the style tag to be the following instead:

``` html
<style is="custom-style">
  body {
    margin: 0;
    padding: 0;
  }
  .notransition {
    --paper-drawer-panel-drawer-container: {
      transition: none !important;
    };
  }
</style>
```

The tests actually passes in ShadyDOM (i.e. improper style scoping), but breaks in ShadowDOM because `.notransition *` doesn't pierce the shadow root. Applying this mixin will, and tests should pass.

**Comment by blasten** [comment]
@danbeam friendly ping.

**Comment by danbeam** [comment]
so, what I don't like about my patch: `rightDrawer` is now `undefined` until `attached()`.
- this could break those doing deep comparisons (i.e. `rightDrawer === true`) before `attached()`
- `rightDrawer` being initially `undefined` might delay template rendering (it's used from various bindings) or cause more re-calculations when it changes (maybe?)

@blasten or @keanulee: thoughts?

additionally, if `rightDrawer` changes in `attached()`, does the newly create element _animatedly_ switch to the right or does it show correctly instantly?  I assume instantly because of the way that templates are asynchronously rendered, but wasn't sure.

**Comment by keanulee** [comment]
@danbeam Yeah, that's not ideal, and it does cause the animation. In `<app-drawer>` (the next iteration of `<paper-drawer-panel>`, we're introducing an `align` property that can be set to start/end, which are RTL compatible, or explicitly to left/right. That would be a breaking API change, so it's hard to add it here.

One suggestion is to try to set `rightDrawer` outside of `<paper-drawer-panel>` based on `document.dir`. It's unfortunate that this element doesn't support this by default, but this is probably the cleanest solution for now. http://jsbin.com/lameliqovi/edit?html,output

**Comment by danbeam** [comment]
Chrome already does [set it externally](https://codereview.chromium.org/1700273003/diff/80001/chrome/browser/resources/settings/settings_ui/settings_ui.js?context=10&column_width=80&tab_spaces=8).

It turns out Chrome's UI sets `<html dir=rtl>` after elements are attached (too late in the process).

That doesn't mean `<paper-drawer-panel>` shouldn't have a better, dynamic default, but the current implementation (and the fact that we must wait until `attached()` to accurately determine `_isRTL`) is kind of a bummer.

I did implement a prototype along the lines of:

``` js
Polymer({
  properties: {
    _rightDrawer: Boolean,
    _defaultDirection: String,

    get rightDrawer() {
      return this._rightDrawer === undefined ? this._defaultDirection == 'rtl' : this._rightDrawer;
    },
    set rightDrawer(rightDrawer) {
      this._rightDrawer = rightDrawer;
    },
  },

  attached: function() {
    this._defaultDirection = window.getComputedStyle(this).direction;
  },
});
```

but didn't find it much better and didn't know how property change accessors would work (maybe switch them all from `(..., rightDrawer, ...)` to `(..., _rightDrawer, _defaultDirection, ...)` or call them programmatically?

at least this way we could determine which thing changed and only animatedly change if `_rightDrawer` was changed?

**Comment by blasten** [comment]
The conclusion here was that the direction should be set externally.

**Comment by blasten** [comment]
@danbeam should we close this PR?

**Pull request closed without merging**